python_ta/debug/id_tracker.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any, Optional
+
+
+class IDTracker:
+    def __init__(self) -> None:
+        self.id_counter: int = 0
+        self.tracker: dict[int, int] = {}
+        self.current_snapshot_objects: set[int] = set()
+
+    def set_id(self, obj) -> Optional[int]:
+        """Assign a unique ID to an object."""
+        if self.is_tracked(obj):
+            return
+        self.id_counter += 1
+        self.tracker[id(obj)] = self.id_counter
+
+    def get_id(self, obj) -> Optional[int]:
+        """Retrieve the unique ID for an object, or none if not tracked."""
+        return self.tracker.get(id(obj), None)
+
+    def is_tracked(self, obj) -> bool:
```

Comment:
Similarly, use `__contains__` here so that you can use the syntax `obj in id_tracker`